src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt

```diff
@@ -228,17 +230,14 @@ class ActivityService(
     val courseActivity = activityRepository.findByIdOrNull(courseActivityId)
       ?: throw NotFoundException("Course Activity $courseActivityId not found")
 
-    if (courseActivity.scheduleEndDate != null && courseActivity.scheduleEndDate!! < date) {
-      throw BadDataException("Course Activity id ${courseActivity.courseActivityId} ended on ${courseActivity.scheduleEndDate}")
-    }
```

Comment:
Don't reject requests where the course ahs already been ended - just continue and try to end any allocations